constellations/orion/src/repos/transaction_repo_test.go

```diff
@@ -0,0 +1,196 @@
+package repos_test
+
+import (
+	"database/sql"
+	"reflect"
+	"testing"
+
+	"github.com/DATA-DOG/go-sqlmock"
+	"github.com/ahsu1230/mathnavigatorSite/constellations/orion/src/domains"
+	"github.com/ahsu1230/mathnavigatorSite/constellations/orion/src/repos"
+	"github.com/ahsu1230/mathnavigatorSite/constellations/orion/src/repos/testUtils"
+)
+
+func initTransactionTest(t *testing.T) (*sql.DB, sqlmock.Sqlmock, repos.TransactionRepoInterface) {
+	db, mock, err := sqlmock.New()
+	if err != nil {
+		t.Fatalf("an error '%s' was not expected when opening a stub database connection", err)
+	}
+	repo := repos.CreateTestTransactionRepo(db)
+	return db, mock, repo
+}
+
+// Test Select All
+func TestSelectAllTransactions(t *testing.T) {
+	db, mock, repo := initTransactionTest(t)
+	defer db.Close()
+
+	// Mock DB statements and execute
+	rows := getTransactionRows()
+	mock.ExpectPrepare("^SELECT (.+) FROM transactions").ExpectQuery().WillReturnRows(rows)
+	got, err := repo.SelectAll()
+	if err != nil {
+		t.Errorf("Unexpected error %v", err)
+	}
+
+	// Validate Results
+	want := []domains.Transaction{getTransaction()}
+	if !reflect.DeepEqual(got, want) {
+		t.Errorf("Values not equal: got = %v, want = %v", got, want)
+	}
+	if err := mock.ExpectationsWereMet(); err != nil {
+		t.Errorf("Unfulfilled expectations: %s", err)
+	}
+}
+
+// Test Select By ID
+func TestSelectTransaction(t *testing.T) {
+	db, mock, repo := initTransactionTest(t)
+	defer db.Close()
+
+	// Mock DB statements and execute
+	rows := getTransactionRows()
+	mock.ExpectPrepare("^SELECT (.+) FROM transactions WHERE id=?").
+		ExpectQuery().
+		WithArgs(1).
+		WillReturnRows(rows)
+	got, err := repo.SelectById(1)
+	if err != nil {
+		t.Errorf("Unexpected error %v", err)
+	}
+
+	// Validate results
+	want := getTransaction()
+	if !reflect.DeepEqual(got, want) {
+		t.Errorf("Values not equal: got = %v, want = %v", got, want)
+	}
+	if err := mock.ExpectationsWereMet(); err != nil {
+		t.Errorf("Unfulfilled expectations: %s", err)
+	}
+}
+
+// Test Insert Transaction
+func TestInsertTransaction(t *testing.T) {
+	db, mock, repo := initTransactionTest(t)
+	defer db.Close()
+
+	// Mock DB statements and execute
+	result := sqlmock.NewResult(1, 1)
+	mock.ExpectPrepare("^INSERT INTO transactions").
+		ExpectExec().
+		WithArgs(
+			sqlmock.AnyArg(),
+			sqlmock.AnyArg(),
+			100,
+			"pay_paypal",
```

Comment:
once you use turn these values into constants, you can access them via `domains.PAY_PAYPAL`.

Now we have a way to ensure people use the correct payment type and less likely to have typos that persist into the database.